akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -28,6 +28,84 @@ import akka.util.OptionVal
 import scala.collection.mutable
 import scala.util.control.NonFatal
 
+@InternalApi
+@ApiMayChange
+private[affinity] object AffinityPool {
+  type PoolState = Int
+  // PoolState: accepts new tasks and processes tasks that are enqueued
+  final val Running = 0
+  // PoolState: does not accept new tasks, processes tasks that are in the queue
+  final val ShuttingDown = 1
+  // PoolState: does not accept new tasks, does not process tasks in queue
+  final val ShutDown = 2
+  // PoolState: all threads have been stopped, does not process tasks and does not accept new ones
+  final val Terminated = 3
+
+  // Method handle to JDK9+ onSpinWait method
+  private val onSpinWaitMethodHandle =
+    try
+      OptionVal.Some(MethodHandles.lookup.findStatic(classOf[Thread], "onSpinWait", methodType(classOf[Unit])))
+    catch {
+      case NonFatal(_) ⇒ OptionVal.None
+    }
+
+  type IdleState = Int
+  // IdleState: Initial state
+  final val Initial = 0
+  // IdleState: Spinning
+  final val Spinning = 1
+  // IdleState: Yielding
+  final val Yielding = 2
+  // IdleState: Parking
+  final val Parking = 3
+
+  // Following are auxiliary class and trait definitions
+  private final class IdleStrategy(val idleCpuLevel: Int) {
+
+    private[this] val maxSpins = 1100 * idleCpuLevel - 1000
```

Comment:
Thanks for pointing this out. @zaharidichev 

I think there is a bug on the calculation of `maxParkNanos`. Since the `idleCpuLevel ` is between 1 and 10 (`1 <= level && level <= 10`), `maxParkNanos`  will be between 250 and -20. And `-20` should not be a proper value for maxParkNanos. 

I can create a PR if this should be fixed.  @patriknw